backend/internal/storage/storage.go

```diff
@@ -0,0 +1,56 @@
+package storage
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+)
+
+type ObjectInfo struct {
+	Path string
+	Size int64
+}
+
+type FileStorage interface {
+	Save(ctx context.Context, relativePath string, data io.Reader) error
+	Open(ctx context.Context, relativePath string) (io.ReadCloser, int64, error)
+	Delete(ctx context.Context, relativePath string) error
+	DeleteAll(ctx context.Context, prefix string) error
+	List(ctx context.Context, prefix string) ([]ObjectInfo, error)
+}
+
+type Config struct {
```

Comment:
Thoughts on having a separate Config for S3 and filesystem? So if in the future new backends are added (Database, Azure Blob...), this doesn't become overloaded